src/WingetCreateCLI/Commands/NewCommand.cs

```diff
@@ -300,12 +334,19 @@ private static T PromptProperty<T>(object model, T property, string memberName,
                     .MakeGenericMethod(property.GetType());
 
                 return (T)generic.Invoke(null, new object[] { message, property.GetType().GetEnumValues(), null, property, null });
-            }
-            else if (typeof(T) != typeof(string) && typeof(IEnumerable<string>).IsAssignableFrom(typeof(T)))
+            }
+            else if (typeof(IEnumerable<string>).IsAssignableFrom(typeof(T)) || typeof(IEnumerable<string>).IsAssignableFrom(typeFromModel))

```

Comment:
> || typeof(IEnumerable<string>).IsAssignableFrom(typeFromModel) [](http://example.com/codeflow?start=76&length=63)

What's a circumstance where this additional check would be required?

Reply:
For optional fields that don't have existing values obtained from the package parser, we can't rely on typeof(T) to determine the type as they would be null and only show a type of System.Object.

The reason why we need both is because of the special case for installerURLs which isn't properly defined in the schema. It was supposed to only be defined as a string. All other List<string> fields are evaluated as lists and not individual strings. This is the rare case where validation actually considers the individual values of the list and not as a whole

Reply:
Discussed offline, there's times when typeof(T) would be Object. So flipping my question, when do we need the first clause here?

Reply:
Makes sense, thanks!

Reply:
The first clause is needed for the InstallerURLs case. If we were to use reflection to determine the type of installerURL it would just be a string and not a List<string>. That is something we created to handle multiple installer urls.  Because we are passing in this.InstallerURLs directly to PromptProperty which is an IEnumerable, this is when the first clause would be true. 

There could be future cases where we start extrapolating out of the schema and start having lists of multiple items similar to the installerUrls.